### PR TITLE
[CHK-2044] feat(ecommerce-for-io): add PM mapping policy for endpoint `GET /transactions/{id}`

### PR DESCRIPTION
**Pull request by giovanniberti**
### List of changes

 * add PM mapping policy for endpoint `GET /transactions/{id}`

### Motivation and context

This PR is part of an effort to map the new eCommerce APIs to the current APIs exposed by Payment Manager (PM).

The transaction id is mapped to a payment context code which is used to retrieve the activatiom status from `pagopa-proxy`. If the response is 200 and a `idPayment` is found, a call to PM's api `/payment/{idPayment}/actions/check` is made and a status of `ACTIVATED` is returned.

Otherwise, a status of `ACTIVATION_REQUESTED` is returned instead.

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<policies>
+    <inbound>
+        <base />
+        <set-variable name="requestTransactionId" value="@{
+            var transactionId = context.Request.MatchedParameters.GetValueOrDefault("transactionId","");
+            if(transactionId == ""){
+                transactionId = context.Request.Headers.GetValueOrDefault("x-transaction-id-from-client","");
```
Comment (pietro-tota) [inline src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl:7]: is this required?

Comment (pietro-tota) [inline src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl:7]: is this required?

File: src/domains/ecommerce-app/env/weu-prod/terraform.tfvars [hunk 2]
```diff
@@ -47,3 +47,10 @@ ecommerce_vpos_psps_list = "BNLIITRR,BCITITMM,UNCRITMM,BPPIITRRXXX,PPAYITR1XXX,B
 ecommerce_npg_psps_list  = ""
 
 dns_zone_checkout = "checkout"
+
+# BEGIN Variables needed for PM <-> eCommerce compatibility layer for IO app
+
+pagopa_proxy_basepath = "CHANGEME"
```
Comment (pietro-tota) [inline src/domains/ecommerce-app/env/weu-prod/terraform.tfvars:53]: are this variables not needed for dev and uat envs?

File: src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl (hunk 1 above)
Comment (giovanniberti) [reply src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl:7]: Nope, removing

File: src/domains/ecommerce-app/env/weu-prod/terraform.tfvars (hunk 2 above)
Comment (giovanniberti) [reply src/domains/ecommerce-app/env/weu-prod/terraform.tfvars:53]: Yep, left from refactoring in this PR, removing

File: src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl (hunk 1 above)
Comment (giovanniberti) [reply src/domains/ecommerce-app/api/ecommerce-io/v1/get_transaction.xml.tpl]: Fixed in a85ef03c

File: src/domains/ecommerce-app/env/weu-prod/terraform.tfvars (hunk 2 above)
Comment (giovanniberti) [reply src/domains/ecommerce-app/env/weu-prod/terraform.tfvars]: Fixed in 1183c80b

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**